src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -97,6 +152,66 @@ struct ExeSpaceUtils<Kokkos::Cuda> {
   static TeamPolicy get_team_policy_force_team_size (Int ni, Int team_size) {
     return TeamPolicy(ni, team_size);
   }
+
+
+
+  template <typename TeamMember, class Lambda, typename ValueType/*, bool Serialize=true*/>
+  static KOKKOS_INLINE_FUNCTION
+  void parallel_reduce (const TeamMember& team,
```

Comment:
This specialization's impl is identical to the general one's impl. So factor this code out into an impl function nested in an `impl` namespace that each can call.